python/ray/serve/tests/test_serve_ha.py

```diff
@@ -109,8 +109,13 @@ def reconfig():
 
     t.join()
 
-    output = worker.exec_run(cmd=f"python -c '{check_script.format(num_replicas=2)}'")
-    assert output.exit_code == 0
+    def check_for_script():
```

Comment:
is it possible to explicitly wait for the head node to be started up instead of blindly retrying the script here? it's possible this could cover up some actual issues